components/common/LeftNav.tsx

```diff
@@ -55,6 +61,21 @@ function LeftNav({
     signInWithGoogle,
     signOutAndClear,
 }: LeftNavProps): JSX.Element {
+    const [role, setRole] = useState<Role | undefined>(undefined)
+
+    const handleSignOut = () => {
+        signOutAndClear()
+        setRole(undefined)
+    }
+
+    useEffect(() => {
+        user?.getIdTokenResult(false).then((data) =>
+            data.claims.role
+                ? setRole(data.claims.role as Role)
+                : setRole(undefined)
+        )
+    }, [user])
```

Comment:
While this probably works, I'd prefer avoiding side effects. We could probably move this one component upwards and pass `role` of type `Role` as a parameter to this component. This would also allow us to drop the `useState` on line 64.

Reply:
Btw, I think we could strive to maximise idiomacy by expressing through types that `User` and `Role` must always exist together: If one does not exist, the other should not exist either. I'd suggest the following.
```typescript
interface UserAndRole {
    user: User
    role: Role
}

interface LeftNavProps {
    userAndRole: UserAndRole | null
    signInWithGoogle: () => Promise<void>
    signOutAndClear: () => Promise<void>
}
```
this could also make the render statement slightly more readible.